packages/airnode-node/src/evm/fulfillments/api-calls.ts

```diff
@@ -75,6 +77,128 @@ async function testFulfill(
   return [[noticeLog], null, goRes.data];
 }
 
+async function estimateGasToFulfill(
+  airnodeRrp: AirnodeRrpV0,
+  request: Request<ApiCallWithResponse>,
+  options: TransactionOptions
+): Promise<LogsErrorData<[ethers.BigNumber, ethers.BigNumber]>> {
+  const noticeLog = logger.pend(
+    'DEBUG',
+    `Attempting to estimate required gas to fulfill API call for Request:${request.id}...`
+  );
+
+  const [estimateGasAirnodeRrpLogs, estimateGasAirnodeRrpErr, estimateGasAirnodeRrpData] =
+    await estimateAirnodeRrpOverhead(airnodeRrp, request, options);
+
+  if (estimateGasAirnodeRrpErr || !estimateGasAirnodeRrpData)
+    return [[noticeLog, ...estimateGasAirnodeRrpLogs], estimateGasAirnodeRrpErr, null];
+
+  const [estimateGasFulfillmentCallLogs, estimateGasFulfillmentCallErr, estimateGasFulfillmentCallData] =
+    await estimateFulfillmentCallOverhead(airnodeRrp, request, options);
+
+  if (estimateGasFulfillmentCallErr || !estimateGasFulfillmentCallData)
+    return [
+      [noticeLog, ...estimateGasAirnodeRrpLogs, ...estimateGasFulfillmentCallLogs],
+      estimateGasFulfillmentCallErr,
+      null,
+    ];
+
+  return [
+    [noticeLog, ...estimateGasAirnodeRrpLogs, ...estimateGasFulfillmentCallLogs],
+    null,
+    [estimateGasAirnodeRrpData, estimateGasFulfillmentCallData],
+  ];
+}
+
+async function estimateAirnodeRrpOverhead(
+  airnodeRrp: AirnodeRrpV0,
+  request: Request<ApiCallWithResponse>,
+  options: TransactionOptions
+): Promise<LogsErrorData<ethers.BigNumber>> {
+  const noticeLog = logger.pend(
+    'DEBUG',
+    `Attempting to estimate AirnodeRrp overhead to fulfill API call for Request:${request.id}...`
+  );
+  if (!request.success) return [[], new Error('Only successful API can be submitted'), null];
+
+  const airnodeRrpDryRunAddress = options.contracts?.AirnodeRrpDryRun;
```

Comment:
Isn't the `?` character not needed here, particularly given `airnodeRrpDryRunAddress!` just below?